### DISCUSSION TITLE
Return infinity when evaluating extremely large numbers

### DISCUSSION THREAD

**Discussion by grdleo** [sympy/sympy#25795] [Q&A]
Hello everyone,
I am building an app with SymPy where the user can enter any expression and can evaluate it using `evalf`.
The expression is parsed using `sympify` using `evaluate=False` parameter. Example:

```python
from sympy import *
>>> expr = sympify("9**9**9**9", evaluate=False)
>>> expr
9**(9**(9**9))
```

And when I call `expr.evalf` the thread (of course) blocks completely, which is a behavior I need to avoid at any cost.
Is there a way to implement some kind of threshold, so that when evaluating the expression, the evaluation just returns infinity?
I was thinking about writing custom `Pow` function with a threshold that checks the arguments and decides if it just returns `oo` instead of doing the calculation. But I would really like to avoid that.

Any ideas? Thank you!

**Comment by oscarbenjamin** [comment]
Unfortunately there are no thresholds for mantissa size in evalf or mpmath. I was going to suggest using lambdify but that still tries to compute the result using `int` for the given example.

**Reply by oscarbenjamin** [reply]
Looks like arb handles this in a better way:
```python
In [1]: import flint

In [2]: nine = flint.arb(9)

In [3]: nine ** nine ** nine ** nine
Out[3]: [+/- inf]
```
